addons/account/wizard/account_move_reversal.py

```diff
@@ -26,8 +26,42 @@ class AccountMoveReversal(models.TransientModel):
             ('modify', 'Full refund and new draft invoice')
         ], string='Credit Method', required=True,
         help='Choose how you want to credit this invoice. You cannot "modify" nor "cancel" if the invoice is already reconciled.')
-    journal_id = fields.Many2one('account.journal', string='Use Specific Journal', help='If empty, uses the journal of the journal entry to be reversed.', check_company=True)
+    journal_id = fields.Many2one(
+        'account.journal',
+        check_company=True,
+        default=lambda self: self._get_default_journal_id(),
+        help='If empty, uses the journal of the journal entry to be reversed.',
+        required=True,
+        string='Use Specific Journal',
+    )
     company_id = fields.Many2one('res.company', required=True, readonly=True)
+    available_journal_ids = fields.Many2many('account.journal', compute='_compute_available_journal_ids')
+
+    @api.depends('move_ids')
+    def _compute_available_journal_ids(self):
+        for record in self:
+            if self.move_ids:
```

Comment:
`self` should be `record`, same everywhere in this method

Reply:
Also, don't split fields like this. File should be ordered as:
- class definition
- default method
- fields
- compute/onchange/business